File: 06_Mon/Cardio.java
import java.util.Scanner;

public class Cardio {
    public static void main(String[] args){
        Integer[] ints = {10,0, 20, 10};
        solution2(ints);
    }

    public static void solution1(Integer[] ints){
        int largest = ints[0];
        int smallest = ints[0];
        for (int i = 0; i < ints.length; i++){
            largest = returnLargest(largest, ints[i]);
            smallest = returnSmallest(smallest, ints[i]);
        }
        System.out.println("Largest : " + largest);
        System.out.println("Smallest : " + smallest);
        System.out.println("Solution : " + (largest - smallest));
    }

    public static void solution2(Integer[] ints){
        int largest = ints[0];
        int smallest = ints[0];
        int i = 0;
        while (i < ints.length){
            largest = returnLargest(largest, ints[i]);
            smallest = returnSmallest(smallest, ints[i]);
            i++;
        }
        System.out.println("Largest : " + largest);
        System.out.println("Smallest : " + smallest);
        System.out.println("Solution : " + (largest - smallest));



    }

    public static int returnLargest(int largest, int currentInt){
        if (largest < currentInt){
            return currentInt;
        }
        return largest;
    }

    public static int returnSmallest(int smallest, int currentInt){
        if (smallest > currentInt){
            return currentInt;
        }
        return smallest;
    }
}

File: 06_Mon/cardio.py
def main():
    ints = [10, 20, 0]
    solution_2(ints)

def solution_1(ints):
    largest = ints[0]
    smallest = ints[0]

    for i in range(0, len(ints)):
        largest = return_largest(largest, ints[i])
        smallest = return_smallest(smallest, ints[i])
    print(str(largest - smallest))

def solution_2(ints):
    largest = ints[0]
    smallest = ints[0]
    i = 0

    while i < len(ints):
        largest = return_largest(largest, ints[i])
        smallest = return_smallest(smallest, ints[i])
        i+=1
    print(str(largest - smallest))
    

def return_largest(largest, current_int):
    if largest < current_int:
        return current_int    
    return largest

def return_smallest(smallest, current_int):
    if smallest > current_int:
        return current_int    
    return smallest

if __name__ == "__main__":
    main()
